core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -566,16 +559,24 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
       connProperties: ConnectionProperties): Option[Any] = {
     SnappyContext.getClusterMode(sparkContext) match {
       case _: ThinClientConnectorMode =>
-        while (!SnappySession.executorAssigned) {
-          // Don't expect this case usually unless lots of applications are submitted in parallel
-          try {
-            Thread.sleep(100)
-          }
-          catch {
-            case _: InterruptedException => logWarning("Interrupted while waiting for executor.")
+        if (!sparkContext.isLocal) {
+          SnappyContext.resourceLock.synchronized {
+            while (!SnappyContext.executorAssigned) {
+              if (!SnappyContext.executorAssigned) {
+                try {
+                  SnappyContext.resourceLock.wait(100)
+                }
+                catch {
+                  case _: InterruptedException =>
+                    logWarning("Interrupted while waiting for executor.")
+                }
+              }
```

Comment:
Should the code also break out of this loop and fail after Property.SerializedWriteLockTimeOut?

Reply:
I guess this is similar to Neeraj's question before. Perhaps you are expecting that lock will timeout and then this can proceed, right?

Reply:
This is a case where even spark just submits the job and is seen stuck on runJob. When an executor gets added to the app, it proceeds. So thinking of keeping similar behavior as spark will keep accepting Application without failure. This will automatically proceed once other App either times out or finishes.